src/android/UploadTask.java

```diff
@@ -457,4 +515,15 @@ public void writeTo(@NonNull BufferedSink bufferedSink) throws IOException {
             }
         }
     }
+
+    private class NetworkReceiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+            if((connectivityManager == null) || (connectivityManager.getActiveNetworkInfo() == null) || (connectivityManager.getActiveNetworkInfo().isConnected() == false)) {
```

Comment:
we can negate the "truthy" condition to make the line shorter

Reply:
okay will test that